### PR TITLE
{filter-bar, device-selector} work

### PR DESCRIPTION
**Pull request by hadim**
- some love to filter bar item when hover
- move old filter-bar-item.styl to device-detector.styl (its original element)

### CODE REVIEW

**Comment by hadim** [comment]
Ok to merge

File: src/coffee/filter-bar/filter-bar-item.styl [hunk 1]
```diff
@@ -1,194 +1,6 @@
...
-
-        img
-            height 20px
-            width 20px
-            position absolute
-            right 3px
-            -webkit-filter: $PngLogo
\ No newline at end of file
+    &:hover
+        background-color #2B3C48
```
Comment (BoLaMN) [inline src/coffee/filter-bar/filter-bar-item.styl]: I would use an opacity level here to keep the possibility of light and dark themes

Comment (hadim) [inline src/coffee/filter-bar/filter-bar-item.styl]: Good idea.

By the way do you know how theme can be handled within angular+material system ?

File: src/coffee/device-selector/device-selector.styl [hunk 2]
```diff
@@ -0,0 +1,194 @@
+.caret
```
Comment (BoLaMN) [inline src/coffee/device-selector/device-selector.styl]: Moving the device-selector over to a md-menu would remove all of this.

Comment (hadim) [inline src/coffee/device-selector/device-selector.styl]: I will try that

File: src/coffee/filter-bar/filter-bar-item.styl (hunk 1 above)
Comment (BoLaMN) [inline src/coffee/filter-bar/filter-bar-item.styl]: The whole app is controlled via the material theming system, due to https://github.com/angular/material/issues/1269 i wrote a custom system on top of it to be able to use the colours everywhere 

https://github.com/butterproject/butter-desktop-angular/blob/master/src/coffee/services/colors.coffee

The variables.styl was basically redundant once all the legacy stylus was removed.

Comment (hadim) [inline src/coffee/filter-bar/filter-bar-item.styl]: Very nice.

So variables.styl can be removed correct ?

Comment (BoLaMN) [inline src/coffee/filter-bar/filter-bar-item.styl]: Not sure if all the legacy stylus was removed but it should of been fairly close to.

**Comment by hadim** [comment]
I also remove .png device icons files and replace them by svg version (for now the svg file contains only the original png file... it would be nice to correctly vectorize icons one day).

File: src/coffee/containers/controls/controls-container.html [hunk 3]
```diff
@@ -1,7 +1,20 @@
-<div class="meta-container-c">
-    <li pt-quality-selector torrents="detail.selected.torrents" selected="detail.currentQuality" class="quality-toggle"></li>
-    <li class="subtitles-dropdown"></li>
-    <li pt-device-selector selected="detail.currentDevice" id="player-option" class="device-dropdown"></li>
+<div class="containers-controls" layout="row" layout-align="center center">
```
Comment (hadim) [inline src/coffee/containers/controls/controls-container.html:1]: angular material layout system seems nice to position this kind of elements. Are you ok ?

File: src/coffee/play-torrent/play-torrent.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<md-button ng-click="startTorrent()" style="color: rgb(255, 255, 255); background-color: rgb(6, 124, 154);" class="watchnow-btn" role="button" tabindex="0">
-  <md-icon md-font-set="material-icons">play_arrow</md-icon> Play
+<md-button class="md-fab" aria-label="Play Button" ng-click="startTorrent()" tabindex="0">
+  <md-icon md-font-set="material-icons" class="ng-scope ng-isolate-scope md-default-theme material-icons">play_arrow</md-icon>
```
Comment (hadim) [inline src/coffee/play-torrent/play-torrent.html]: I replaced by a circle button

File: src/coffee/quality-selector/quality-selector.coffee [hunk 5]
```diff
@@ -9,14 +9,21 @@ angular.module 'app.quality-selector', []
...
   controller: 'qualityCtrl as quality'
 
-.controller 'qualityCtrl', ->
+.controller 'qualityCtrl', ($scope) ->
   vm = this
 
-  vm.list = ['480p', '720p', '1080p']
+  # I dont knwy but in the template, quality.torrents is empty...
+  # So I fake it for now...
+  vm.fakelist = {"480p": "blablab", "720p": "blablab", "1080p": "blablab"}
```
Comment (hadim) [inline src/coffee/quality-selector/quality-selector.coffee:17]: quality.torrents is empty.... any toughs ?

File: src/coffee/quality-selector/quality-selector.html [hunk 6]
```diff
@@ -1 +1,17 @@
-<span ng-repeat="(q, torrent) in quality.torrents" ng-if="q !== '0'" class="q{{ q }}" ng-class="{ 'active': quality.torrents[quality.selected].url === quality.torrents[q].url }" ng-click="quality.select(q)">{{ q }}</span>
\ No newline at end of file
+<md-menu>
```
Comment (hadim) [inline src/coffee/quality-selector/quality-selector.html:1]: Use md-menu for quality-selector

**Comment by hadim** [comment]
@BoLaMN Comments ?

File: src/coffee/play-torrent/play-torrent.html (hunk 4 above)
Comment (BoLaMN) [inline src/coffee/play-torrent/play-torrent.html]: Ng-scope and Ng-isolated-scope?

File: src/coffee/quality-selector/quality-selector.coffee (hunk 5 above)
Comment (BoLaMN) [inline src/coffee/quality-selector/quality-selector.coffee:17]: Prob due to the api change

File: src/coffee/containers/controls/controls-container.html (hunk 3 above)
Comment (BoLaMN) [inline src/coffee/containers/controls/controls-container.html:1]: Yeah was on the todo list to convert it all over and remove the absolute positioning.

**Comment by BoLaMN** [comment]
With the svg icons bring them in thru $mdIconProvider for easier reuse https://material.angularjs.org/latest/api/service/$mdIconProvider

**Comment by hadim** [comment]
Ok I will do that but can we merge this before ? So I can open a new PR.

**Comment by hadim** [comment]
This one is going to be too big at one moment !

**Comment by BoLaMN** [comment]
Small still ;) 

I have no say over PRs as I'm not actually an active dev on this lol. I'm just offering suggestions and hints as I wrote most of the current angular version for fun in my spare time so I kinda have an idea how it all works. 

I'll probably continue building the webchimera player as I ripped most of that out in to a seperate module to make it easier to work on the player.

**Comment by hadim** [comment]
Ok thank you for that @BoLaMN. I may ping you sometime to get suggestions from you.

To be sure, I'd like to know if im not working for nothing here. Is angular and this repo going to be the main butter app one day (even if we are far from this) or is this just a proff of concept no one care ?

Once you have a player working @BoLaMN , merge it asap here so we can move forward !

**Comment by BoLaMN** [comment]
Sure happy to offer suggestions.

The newer player module lives at https://github.com/BoLaMN/wcjs-angular and basically 95% of the webchimera folder could be removed in place of it if you use it.

**Comment by hadim** [comment]
Ok so do you suggest that wcjs-angular would live as a separate project and we just get it in butter with npm or are you planning to merge it here at one moment ?

**Comment by BoLaMN** [comment]
I'd suggest a seperate module as it makes working on either project easier. the player module is pretty universal and allows you to extend via your own plugins for what ever butters needs are.

**Comment by hadim** [comment]
Sounds good to me ! Feel free to submit PR here to include wcjs-angular as a dependency and the player.

What about the streamer ? Is it working ?

**Comment by BoLaMN** [comment]
The streamers all in the server folder. 

my end goal was to decouple the browser rendering from the node processes to not block the ui but  yes it was working previously.

**Pull request merged**